pgx-version-updater/src/main.rs

```diff
@@ -356,51 +361,50 @@ fn is_cargo_toml_file(entry: &DirEntry) -> bool {
     let mut result = String::new();
 
-    if old_version_specifier.chars().nth(0).unwrap().is_numeric() {
-        result.push_str(old_version_specifier);
-    } else {
-        let version_pos = old_version_specifier
-            .find(|c: char| c.is_numeric())
-            .unwrap();
-
-        result.push_str(&old_version_specifier[..version_pos]);
-        result.push_str(&new_version.clone());
+    match current_version_specifier.chars().nth(0) {
+        // If first character is numeric, then we have just a version specified,
+        // such as "0.5.2" or "4.15.0"
+        Some(c) if c.is_numeric() => result.push_str(current_version_specifier),
+
+        // Otherwise, we have a specifier such as "=0.5.2" or "~0.4.6" or ">= 1.2.0"
+        // Extract out the non-numeric prefix and join it with the new version to
+        // be used. e.g. "=0.5.2" to new version "0.5.4" would result in "=0.5.4"
+        // TODO: This does not currently handle any specifiers with wildcards,
+        // such as "1.*"
+        Some(_) => {
+            if let Some(version_pos) = current_version_specifier.find(|c: char| c.is_numeric()) {
+                result.push_str(&current_version_specifier[..version_pos]);
+                result.push_str(&new_version.clone());
+            } else {
+                panic!(
+                    "Could not find an actual version in specifier: '{}'",
+                    current_version_specifier
+                );
+            }
+        }
+        None => panic!("Version specifier '{}' is not valid!", current_version_specifier),
     }
 
     result
 }
 
 // Given a filepath pointing to a Cargo.toml file, extract out the [package] name
 // if it has one
-fn extract_package_name(filepath: &String) -> Option<String> {
+fn extract_package_name<P: AsRef<Path>>(filepath: P) -> Option<String> {
+    let filepath = filepath.as_ref();
+
     let data = fs::read_to_string(&filepath)
-        .expect(format!("Unable to open file at {}", &filepath).as_str());
+        .expect(format!("Unable to open file at {}", &filepath.display()).as_str());
+
     let doc = data.parse::<Document>().expect(
-        format!(
-            "File at location {} is an invalid Cargo.toml file",
-            &filepath
-        )
-        .as_str(),
+        format!("File at location {} is an invalid Cargo.toml file", &filepath.display()).as_str(),
     );
 
-    if doc.contains_key("package") {
-        let package_table = doc.get("package").unwrap().as_table().unwrap();
-
-        if package_table.contains_key("name") {
-            let package_name = package_table
-                .get("name")
-                .unwrap()
-                .as_str()
-                .unwrap()
-                .to_string();
-
-            Some(package_name)
-        } else {
-            None
-        }
+    if let Some(table) = doc.get("package").and_then(|i| i.as_table()) {
+        table.get("name").and_then(|i| i.as_str().and_then(|i| Some(i.to_string())))
```

Comment:
when you have to drill down like whoa

Reply:
I know, I hate the look of it. Is there any way to make it more concise? I think I am somewhat limited to the API of `toml_edit`

Reply:
Yes, but it looks like...